WebDevStudios/wd_s
# Theme/#408 remove inline styles

Closes #408 

### DESCRIPTION ###
This replaces all of our inline styles and, in doing so, gets us more compliant with Section 508 and WCAG 2.0 AA. Neat!

### Color Picker
We went to great lengths in the past to customize the ACF Color Picker field to only show the brand/theme colors we specified while hiding the color picker gradient itself. While this made sense in a general sense, it seemed like kind of overkill to use the Color Picker field if we're just going to hide the actual picker.

This removes our usage of the Color Picker from Background and Other Options in favor of a standard `select` field which we filter via a PHP function in the theme's files.
![](https://dl.dropbox.com/s/ppip5ucbkszuhcq/Screenshot%202018-10-05%2014.28.58.jpg?dl=0)

Instead of adding these values as an inline style, they now get added as classes like `background-blue` for background colors and `text-cod-gray` for text colors.

### Image Background
The image background was the other item still being used as an inline style. This update moves away from that and displays the image as a `<figure>` in the same way we're displaying Video Backgrounds with positioning and z-indexing to place it behind the content while centering it vertically and horizontally.
![](https://dl.dropbox.com/s/jyx93osyn1wfynx/Screenshot%202018-10-05%2014.31.55.jpg?dl=0)

### Video Background
No changes here. I've combined some styles with new styles for the image background, but have not added any new styles for functionality for the video background.

### Global Content Block Renaming
I renamed the "Block Options" field group to "Other Options". The tab itself on our blocks is "Other Options", so I renamed the field group to match.

### Total Validator
This gets rid of all of the dings against our use of inline styles. Any remaining warnings or issues have to do with empty `alt` tags and TVP disliking WP's use of the `sizes` attribute on `img` tags within these blocks.

### SCREENSHOTS ###
**Desktop**
![](https://dl.dropbox.com/s/wb9x4qsyvlo3tc1/screencapture-cmc-underscores-color-picker-testing-2018-10-05-14_04_56.jpg?dl=0)

**Mobile**
![](https://dl.dropbox.com/s/axm6z8owzsrtwap/screencapture-cmc-underscores-color-picker-testing-2018-10-05-14_07_17.jpg?dl=0)

### OTHER ###

### STEPS TO VERIFY ###
Check out the branch and add any combination of blocks, testing the usage of background images, background colors, and background videos as well as text colors.

### DOCUMENTATION ###
Yes – we'll need to update a screenshot on this page if this is accepted: https://github.com/WebDevStudios/wd_s/wiki/ACF-Content-Blocks#other-options
